src/inputs/selectNext/selectNext.tsx

```diff
@@ -0,0 +1,439 @@
+                autoHeightMin={selectProps.dropdownMenuMinHeight}
+                autoHide
+                className="select-menu-scrollbar"
+                ref={selectProps.menuScrollBarRef}
+            >
+                {children}
+            </ScrollBar>
+        </components.MenuList>
+    );
+};
+
+const CustomOption = (componentProps) => {
+    const {
+        children,
+        className,
+        innerRef,
+        isSelected,
+        selectOption,
+    } = componentProps;
+
+    const optionClass = ClassNames(
+        className,
+    );
+
+    return (
+        <components.Option
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...componentProps}
+            aria-selected={isSelected}
+            className={optionClass}
+            selectOption={selectOption}
+            ref={innerRef}
+            tabIndex={0}
+        >
+            {children}
+        </components.Option>
+    );
+};
+
+type Option = {
+    value: string;
+    label: string;
+};
+
+/**
+ * The Select component represents a control that provides a menu of options.
+ */
+// eslint-disable-next-line prefer-arrow-callback
+const SelectNext = React.forwardRef(function SelectNext(
+    props: PropTypes, ref: LegacyRef<HTMLDivElement>,
+) {
+    const {
+        className,
+        dropdownMenuMaxHeight,
+        dropdownMenuMinHeight,
+        id,
+        isSearchable,
+        label,
+        menuPortalTarget,
+        noOptionsMessage,
+        onChange: onChangeProp,
+        options,
+        placeholder,
+        styles,
+        tabIndex,
+        value,
+        ...otherProps
+    } = props;
+
+    const classes = useStyles(props);
+
+    const onChange = (selectedOption) => {
+        if (isFunction(onChangeProp)) {
+            onChangeProp(selectedOption);
+        }
+    };
+
+    const rootClasses = ClassNames(
+        UI_CLASS_NAME,
+        BEM_SELECT,
+        classes.root,
+        className,
+    );
+
+    return (
+        <div
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...otherProps}
+            className={rootClasses}
+            id={id}
+            ref={ref}
+        >
+            {label && (
+                <label
+                    className={ClassNames(
+                        'label',
+                        classes.label,
+                    )}
+                >
+                    {label}
+                </label>
+            )}
+
+            <Select
+                components={{
+                    DropdownIndicator: CustomArrow,
+                    MenuList: CustomMenuList,
+                    Option: CustomOption,
+                }}
+                // @ts-ignore
+                dropdownMenuMaxHeight={dropdownMenuMaxHeight}
+                dropdownMenuMinHeight={dropdownMenuMinHeight}
+                isSearchable={isSearchable}
+                menuPortalTarget={menuPortalTarget}
+                name="cm_ui_select_next"
```

Comment:
Do we want to hard-code a name?  That needs to be settable via prop, no?
Like if consumer has a form with multiple selects, don't they need distinct names?

Reply:
Defined via props!